modules/internal/ChapelArray.chpl

```diff
@@ -2037,7 +2061,18 @@ module ChapelArray {
             hi = this.domain.high+1;
       const newDom = {lo..hi};
       on this._value {
-        this._value.dsiReallocate(newDom);
```

Comment:
I take back my previous comment about 'newDom'.  I see what you mean now and can clean that up in another PR.